src/Analyzers/CSharp/CodeFixes/UseUTF8StringLiteral/UseUTF8StringLiteralCodeFixProvider.cs

```diff
@@ -0,0 +1,191 @@
+                var stringValue = GetUTF8StringValueForDiagnostic(semanticModel, diagnostic, cancellationToken);
+
+                // If we're replacing a byte array that is passed to a parameter array, not and an explicit array creation
+                // then node will be the ArgumentListSyntax that the implicit array creation is just a part of, so we have
+                // to handle that separately, as we can't just replace node with a string literal
+                //
+                // eg given a method:
+                //     M(string x, params byte[] b)
+                // our diagnostic would be reported on:
+                //     M("hi", [|1, 2, 3, 4|]);
+                // but node will point to:
+                //     M([|"hi", 1, 2, 3, 4|]);
+
+                if (node is BaseArgumentListSyntax argumentList)
+                {
+                    editor.ReplaceNode(node, CreateArgumentListWithUTF8String(argumentList, diagnostic.Location, stringValue));
+                }
+                else
+                {
+                    editor.ReplaceNode(node, CreateUTF8String(node, stringValue));
+                }
+            }
+        }
+
+        private static string GetUTF8StringValueForDiagnostic(SemanticModel semanticModel, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            // For computing the UTF8 string we need the original location of the array creation
+            // operation, which is stored in additional locations.
+            var location = diagnostic.AdditionalLocations[0];
+            var node = location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+
+            var operation = semanticModel.GetRequiredOperation(node, cancellationToken);
+
+            // Because we get the location from an IOperation.Syntax, sometimes we have to look a
+            // little harder to get back from syntax to the operation that triggered the diagnostic
+            if (operation is not IArrayCreationOperation arrayOp)
+            {
+                if (node is LiteralExpressionSyntax)
+                {
+                    // For single literals, that likely means a collection initializer, and the array creation
+                    // will be a parent of the operation
+                    arrayOp = FindArrayCreationOperationAncestor(operation);
+                }
+                else
+                {
+                    // Otherwise, we must have an implicit array creation for a parameter array, so the location
+                    // will be the invocation, or similar, that has the argument, and we need to descend child
+                    // nodes to find the one we are interested in. To make sure we're finding the right one,
+                    // we can use the diagnostic location for that, since the analyzer raises it on the first element.
+                    arrayOp = operation.DescendantsAndSelf()
```

Comment:
check this on the analyze side as welll.  you say we must have a parameter-array, but i'm not at all confident unless the analyzer verified that.